src/MPAGSCipher/ProcessCommandLine.hpp

```diff
@@ -1,12 +1,18 @@
-#ifndef MPAGSCIPHER_PROCESSCOMMANDLINE_HPP
-#define MPAGSCIPHER_PROCESSCOMMANDLINE_HPP
-
-#include <string>
+#ifndef MPAGSCIPHER_PROCESSCMDLINE_HPP
+#define MPAGSCIPHER_PROCESSCMDLINE_HPP
 #include <vector>
+#include <string>
+using namespace std;
 
-bool processCommandLine(const std::vector<std::string>& cmdLineArgs,
-                        bool& helpRequested, bool& versionRequested,
-                        std::string& inputFile, std::string& outputFile,
-                        std::string& cipherKey, bool& encrypt);
+bool processCmdLine(const vector<string>& args,
+                    bool& helpRequested,
+                    bool& versionRequested,
+                    string& inputFileName,
+                    string& outputFileName,
+                    bool& encrypt,
+                    bool& decrypt,
+                    bool& keySupplied,
+                    int& key);
```

Comment:
Part of this exercise was to introduce a struct to combine the settings in order to reduce the number of parameters of this function:
```c++
struct ProgramSettings {
    /// Indicates the presence of the help flag in the arguments
    bool helpRequested;
    /// Indicates the presence of the version flag in the arguments
    bool versionRequested;
    /// Name of the input file
    std::string inputFile;
    /// Name of the output file
    std::string outputFile;
    /// Key to be used in encrypting/decrypting routine
    std::string cipherKey;
    /// Flag indicating the mode in which the cipher should run (i.e. encrypt or decrypt)
    CipherMode cipherMode;
    /// Flag indicating which cipher to use (e.g. Caesar, Playfair, etc.)
    CipherType cipherType;
};
```
With this the function becomes:
```c++
bool processCommandLine(const std::vector<std::string>& cmdLineArgs,
                        ProgramSettings& settings);
```